pydantic/typing.py

```diff
@@ -80,8 +81,21 @@ def evaluate_forwardref(type_: ForwardRef, globalns: Any, localns: Any) -> Any:
         except ImportError:
             Literal = None
 
-    def get_args(t: Type[Any]) -> Tuple[Any, ...]:
-        return getattr(t, '__args__', ())
+    if sys.version_info < (3, 7):
+
+        def get_args(t: Type[Any]) -> Tuple[Any, ...]:
+            return getattr(t, '__args__', ())
+
+    else:
+        _GenericAlias = getattr(typing, '_GenericAlias')
+
+        def get_args(t: Type[Any]) -> Tuple[Any, ...]:
```

Comment:
please rebase to master and check this is still required.

Reply:
Yes, this is already rebased.  Unfortunately, typing changes a lot in between 3.6 and 3.9.
For python `< 3.8`, we need to vendor `get_args` and do it correctly https://github.com/python/cpython/blob/4f3c25043d651a13c41cffcee703f7d5cb677cc7/Lib/typing.py#L1497 (mainly for all the generics stuff), but to do this correctly we need `_GenericAlias` which doesn't exist in python 3.6.
Since generics don't support 3.6 anyway that's ok for us (we just use the simpler version of `return getattr(t, '__args__', ())`, and for python >= 3.8 we can just use the cpython version. 
So unfortunately we need all three cases.

I'll add comments though explaining this